load/scan.go

```diff
@@ -5,12 +5,12 @@ import (
 	"reflect"
 )
 
-// ackAndMaybeSend adjust the outstanding batches count and potentially sends
-// another batch to the worker via ch. If unsent is non-empty, send. Returns
-// the updated state of unsent
-func ackAndMaybeSend(ch *duplexChannel, count *int, unsent []Batch) []Batch {
+// sendBatch adjust the outstanding batches count
+// and sends one batch (if any available) to the worker via ch.
+// Returns the updated state of unsent
+func sendBatch(ch *duplexChannel, count *int, unsent []Batch) []Batch {
```

Comment:
So I still prefer the old name because it is more clear what the function does. It acknowledges that receipt of a 'done' message and possibly (maybe) sends if there is one to send. Without reading the godoc on this, one may get the incorrect impression of what it's doing.

Reply:
Ok, I agree, `ackAndMaybeSend` would be more descriptive in this case. It were my attempt to understand code through refactoring. Updated/reverted accordingly